modules/core/include/opencv2/core/dualquaternion.inl.hpp

```diff
@@ -387,6 +392,94 @@ DualQuat<T> DualQuat<T>::sclerp(const DualQuat<T> &q0, const DualQuat<T> &q1, co
     return v0 * v0inv1.power(t, QUAT_ASSUME_UNIT);
 }
 
+template <typename T>
+DualQuat<T> DualQuat<T>::dqblend(const DualQuat<T> &q1, const DualQuat<T> &q2, const T t, QuatAssumeType assumeUnit)
+{
+    DualQuat<T> v1(q1), v2(q2);
+    if (!assumeUnit)
+    {
+        v1 = v1.normalize();
+        v2 = v2.normalize();
+    }
+    if (v1.getRotation().dot(v2.getRotation()) < 0)
+    {
+        return ((1 - t) * v1 - t * v2).normalize();
+    }
+    return ((1 - t) * v1 + t * v2).normalize();
+}
+
+template <typename T>
+DualQuat<T> DualQuat<T>::gdqblend(const std::vector<DualQuat<T>> &_dualquat, const std::vector<T> &weight, QuatAssumeType assumeUnit)
+{
+    std::vector<DualQuat<T>> dualquat(_dualquat);
+    if (!assumeUnit)
+    {
+        for (auto &dq : dualquat)
+        {
+            dq = dq.normalize();
+        }
+    }
+    size_t cn = dualquat.size();
+    if (cn == 0)
+    {
+        return DualQuat<T>(1, 0, 0, 0, 0, 0, 0, 0);
+    }
+    DualQuat<T> dq_blend = dualquat[0] * weight[0];
+    Quat<T> q0 = dualquat[0].getRotation();
+    for (size_t i = 1; i < cn; ++i)
+    {
+        T k = q0.dot(dualquat[i].getRotation()) < 0 ? -1: 1;
+        dq_blend = dq_blend + dualquat[i] * k * weight[i];
+    }
+    return dq_blend.normalize();
+}
+
+template <typename T>
+void DualQuat<T>::dqs(const std::vector<Vec<T, 3>> &in_vert, const std::vector<Vec<T, 3>> &in_normals,
+         std::vector<Vec<T, 3>> &out_vert, std::vector<Vec<T, 3>> &out_normals,
+         const std::vector<DualQuat<T>> &_dualquat, const std::vector<std::vector<T>> &weights,
+         const std::vector<std::vector<int>> &joint_id, QuatAssumeType assumeUnit)
+{
```

Comment:
I think this function should be moved to some .cpp file. Here's motivation why:
Typical use case for this function is to transform point clouds and vertices which are counted in thousands. This may involve massive parallel processing using `parallel_for_`, SIMD or OpenCL.
Having such a code in a header makes no sense; this would slow down the compilation. Having this in `.inl.hpp` has even less sense since this is not a good function to be inlined.

Reply:
OK,  I've temporarily removed this function,  but have no idea where to put it

Reply:
You can create a separate .cpp file, like `dqs.cpp` or something

Reply:
OK

Reply:
Is it possible to separate declarations and definitions of template functions? I did some consulting and I saw that most of the template class functions that can be run were written together. Is there a way to separate them?